platform/harmony/api.go

```diff
@@ -0,0 +1,99 @@
+package harmony
+
+import (
+	"github.com/spf13/viper"
+	"github.com/trustwallet/blockatlas/coin"
+	"github.com/trustwallet/blockatlas/pkg/blockatlas"
+	"strconv"
+)
+
+type Platform struct {
+	client Client
+}
+
+func (p *Platform) Init() error {
+	p.client = Client{blockatlas.InitClient(viper.GetString("harmony.rpc"))}
+	return nil
+}
+
+func (p *Platform) Coin() coin.Coin {
+	return coin.Coins[coin.ONE]
+}
+
+func (p *Platform) GetTxsByAddress(address string) (blockatlas.TxPage, error) {
+	result, err := p.client.GetTxsOfAddress(address)
+	if err != nil {
+		return []blockatlas.Tx{}, err
+	}
+	return NormalizeTxs(result.Transactions), err
+}
+
+func (p *Platform) CurrentBlockNumber() (int64, error) {
+	return p.client.CurrentBlockNumber()
+}
+
+func (p *Platform) GetBlockByNumber(num int64) (*blockatlas.Block, error) {
+	if srcBlock, err := p.client.GetBlockByNumber(num); err == nil {
+		block := p.NormalizeBlock(&srcBlock)
+		return &block, nil
+	} else {
+		return nil, err
+	}
+}
+
+func (p *Platform) NormalizeBlock(block *BlockInfo) blockatlas.Block {
+	blockNumber, err := hexToInt(block.Number)
+	if err != nil {
+		return blockatlas.Block{}
+	}
+	return blockatlas.Block{
+		ID:     block.Hash,
+		Number: int64(blockNumber),
+		Txs:    NormalizeTxs(block.Transactions),
+	}
+}
+
+func NormalizeTxs(txs []Transaction) []blockatlas.Tx {
+	normalizeTxs := make([]blockatlas.Tx, 0)
+	for _, srcTx := range txs {
+		normalized, isCorrect := NormalizeTx(&srcTx)
+		if !isCorrect {
+			return []blockatlas.Tx{}
+		}
+		normalizeTxs = append(normalizeTxs, normalized)
+	}
+	return normalizeTxs
+}
+
+func NormalizeTx(trx *Transaction) (tx blockatlas.Tx, b bool) {
+	gasPrice, err := hexToInt(trx.GasPrice)
+	gas, err := hexToInt(trx.Gas)
```

Comment:
You need to handle this error
```
if err != nil {
	return blockatlas.Tx{}, false
}
```

Reply:
In the `NormalizeTx` function there are a lot of `err` assignments, there's a way to avoid 

```go
if err != nil {
    return blockatlas.Tx{}, false
}
```

after each line? Otherwise I'll go with duplication

Reply:
Yes, you need to handle all errors, it's annoying, I know, but is the Go pattern, you always need to handle an error, and try to put all possible information inside the error, we have our own error package to do that `github.com/trustwallet/blockatlas/pkg/errors`

Example:
https://github.com/trustwallet/blockatlas/blob/ee5e423df6adf6b431b2aa0460d8becb95d6f53b/pkg/blockatlas/client.go#L63

You can push this error to sentry if you add the method in the end:`.PushToSentry()`

Reply:
Great! Thank you for the explanation, if that's the correct way I'll do this without further ado